service/src/main/java/se/inera/statistics/service/DeleteCustomerDataImpl.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2022 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.statistics.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import se.inera.statistics.service.warehouse.db.DeleteCustomerDataDB;
+import se.inera.statistics.service.warehouse.db.MeddelandehandelseMessagewidelineResultDao;
+
+@Service
+public class DeleteCustomerDataImpl implements DeleteCustomerData {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DeleteCustomerDataImpl.class);
+
+    private DeleteCustomerDataDB deleteCustomerDataDB;
+
+    @Autowired
+    public DeleteCustomerDataImpl(DeleteCustomerDataDB deleteCustomerDataDB) {
+        this.deleteCustomerDataDB = deleteCustomerDataDB;
+    }
+
+    @Override
+    public List<DeleteCustomerDataByIntygsIdDao> deleteCustomerDataByIntygsId(List<String> intygsIdList) {
```

Comment:
If the db-delete throws an error the method should not return the "intygsId" as deleted.

Reply:
Fixed!